commands/diagnostics.vegman

```diff
@@ -119,3 +119,57 @@ function cmd_traceroute {
 function cmd_netstat {
   netstat "$@"
 }
+
+# D-Bus intrusion objects
+INTR_SERVICE="xyz.openbmc_project.intrusionsensor.service"
+INTR_GPIO_PIN="CASE_OPEN_FAULT_RST"
+INTR_DBUS_SERVICE="xyz.openbmc_project.IntrusionSensor"
+INTR_DBUS_PATH="/xyz/openbmc_project/Intrusion/Chassis_Intrusion"
+INTR_DBUS_INTERFACE="xyz.openbmc_project.Chassis.Intrusion"
+INTR_DBUS_PROPERTY="Status"
+
+# @sudo cmd_intrusion_on admin,tech
+# @restrict cmd_intrusion_on admin,tech
+# @doc cmd_intrusion_on
+# Turn the chassis intrusion detection on
+function cmd_intrusion_on {
+  sysctemctl start ${INTR_SERVICE}
+}
+
+# @sudo cmd_intrusion_off admin,tech
+# @restrict cmd_intrusion_off admin,tech
+# @doc cmd_intrusion_off
+# Turn the chassis intrusion detection off
+function cmd_intrusion_off {
+  sysctemctl stop ${INTR_SERVICE}
+}
+
+# @sudo cmd_intrusion_show admin,tech
+# @restrict cmd_intrusion_show admin,tech
+# @doc cmd_intrusion_show
+# Show the chassis intrusion detection status
+function cmd_intrusion_show {
+  state=$(busctl get-property ${INTR_DBUS_SERVICE} "${INTR_DBUS_PATH}" \
+          ${INTR_DBUS_INTERFACE} ${INTR_DBUS_PROPERTY} \
+          | awk '{print $NF;}' 2>/dev/null ||:)
+  if [ "$state" == "Normal" ]; then
+    echo "Chassis is closed"
+  else
+    echo "Hardware intrusion detected"
+  fi
+}
+
+# @sudo cmd_intrusion_clear admin,tech
+# @restrict cmd_intrusion_clear admin,tech
+# @doc cmd_intrusion_clear
+# Clear the chassis intrusion status
+function cmd_intrusion_clear {
+  # GPIO pin A1 = 1
+  pin=$(gpiofind ${INTR_GPIO_PIN} | awk '{print $NF;}')
+  if [[ -n "${pin}" ]]; then
+    gpioset gpiochip0 -m wait -u 250000 "${pin}"=1 && \
```

Comment:
And as this operation has high security matter, it probably has a sense to prepend this with user confirmation and an optional key `-y/--yes` for using in the batch mode.